dart/lib/src/sentry_transaction.dart

```diff
@@ -0,0 +1,137 @@
+import '../sentry.dart';
+import 'sentry_tracer.dart';
+import 'utils.dart';
+
+class SentryTransaction extends SentryEvent {
+  late final DateTime startTimestamp;
+  static const String _type = 'transaction';
+  late final List<ISentrySpan> spans;
+  final SentryTracer _tracer;
+
+  SentryTransaction(
+    this._tracer, {
+    SentryId? eventId,
+    DateTime? timestamp,
+    String? platform,
+    String? serverName,
+    String? release,
+    String? dist,
+    String? environment,
+    String? transaction,
+    dynamic throwable,
+    Map<String, String>? tags,
+    Map<String, dynamic>? extra,
+    SentryUser? user,
+    Contexts? contexts,
+    List<Breadcrumb>? breadcrumbs,
+    SdkVersion? sdk,
+    SentryRequest? request,
+    String? type,
+  }) : super(
+          eventId: eventId,
+          timestamp: timestamp ?? _tracer.endTimestamp,
+          platform: platform,
+          serverName: serverName,
+          release: release,
+          dist: dist,
+          environment: environment,
+          transaction: transaction ?? _tracer.name,
+          throwable: throwable,
+          tags: tags ?? _tracer.context.tags,
+          extra: extra ?? _tracer.data,
+          user: user,
+          contexts: contexts,
+          breadcrumbs: breadcrumbs,
+          sdk: sdk,
+          request: request,
+          type: _type,
+        ) {
+    startTimestamp = _tracer.startTimestamp;
+
+    final spanContext = _tracer.context;
+    // this.spans = spans ?? _tracer.children;
+    spans = _tracer.children;
+
+    final traceContext = SentryTraceContext(
+      operation: spanContext.operation,
+      traceId: spanContext.traceId,
+      spanId: spanContext.spanId,
+      description: spanContext.description,
+      status: spanContext.status,
+      parentSpanId: spanContext.parentSpanId,
+      sampled: spanContext.sampled,
+    );
+
+    this.contexts.trace = traceContext;
```

Comment:
should we allow to mutate the trace context? right now I don't, I always set the trace context from `_tracer` when using `copyWith`

Reply:
dont expose it but assign the value of contexts.trace directly instead of creating a new SentryTraceContext which isn't scalable when adding new fields